From b794c685686eefbc6144104611385f3f098bdf71 Mon Sep 17 00:00:00 2001
From: Karol1223 <68503002+Karol1223@users.noreply.github.com>
Date: Sun, 18 Aug 2024 02:28:06 +0200
Subject: [PATCH 041/125] [TropiCata] Flying bird audit (#75769)

* obsoletion

* toucan

* parakeet

* macaw

* useless field

* saffron finch + egg mongroup cleanup

* kiskadee

* hornero

* description updates

* Update dictionary.txt
---
 .../TropiCataclysm/items/comestibles/egg.json |  59 ++--
 .../TropiCataclysm/monstergroups/eggs.json    |  95 +++---
 .../monstergroups/wilderness.json             |  31 +-
 .../monsters/tropical_bird.json               | 287 +++++-------------
 .../migration_items.json                      |  17 ++
 data/mods/TropiCataclysm/requirements.json    |  18 ++
 tools/spell_checker/dictionary.txt            |   5 +
 7 files changed, 194 insertions(+), 318 deletions(-)
 create mode 100644 data/mods/TropiCataclysm/obsoletion_and_migration_0.I/migration_items.json
 create mode 100644 data/mods/TropiCataclysm/requirements.json

diff --git a/data/mods/TropiCataclysm/items/comestibles/egg.json b/data/mods/TropiCataclysm/items/comestibles/egg.json
index 4425204b52..ace06780d1 100644
--- a/data/mods/TropiCataclysm/items/comestibles/egg.json
+++ b/data/mods/TropiCataclysm/items/comestibles/egg.json
@@ -27,42 +27,47 @@
   {
     "type": "COMESTIBLE",
     "id": "egg_toucan",
-    "name": "toucan egg",
-    "copy-from": "egg_bird",
-    "rot_spawn": "GROUP_EGG_TOUCAN",
-    "rot_spawn_chance": 70
+    "name": { "str": "toco toucan egg" },
+    "spoils_in": "9 days",
+    "copy-from": "egg_crow",
+    "rot_spawn": "GROUP_EGG_TOUCAN"
   },
   {
     "type": "COMESTIBLE",
     "id": "egg_parrot",
-    "name": "parrot egg",
-    "copy-from": "egg_bird",
-    "rot_spawn": "GROUP_EGG_PARROT",
-    "rot_spawn_chance": 70
+    "name": { "str": "white-eyed parakeet egg" },
+    "spoils_in": "13 days",
+    "copy-from": "egg_bluejay",
+    "rot_spawn": "GROUP_EGG_PARROT"
   },
   {
     "type": "COMESTIBLE",
-    "id": "egg_budgerigar",
-    "name": "budgerigar egg",
-    "copy-from": "egg_bird",
-    "rot_spawn": "GROUP_EGG_BUDGERIGAR",
-    "rot_spawn_chance": 70
+    "id": "egg_macaw",
+    "name": { "str": "blue-and-yellow macaw egg" },
+    "spoils_in": "13 days",
+    "copy-from": "egg_crow",
+    "rot_spawn": "GROUP_EGG_MACAW"
   },
   {
     "type": "COMESTIBLE",
-    "id": "egg_cockatoo",
-    "name": "cockatoo egg",
-    "copy-from": "egg_bird",
-    "rot_spawn": "GROUP_EGG_COCKATOO",
-    "rot_spawn_chance": 70
+    "id": "egg_finch_saffron",
+    "name": { "str": "saffron finch egg" },
+    "copy-from": "egg_waxwing",
+    "rot_spawn": "GROUP_EGG_FINCH_SAFFRON"
   },
   {
     "type": "COMESTIBLE",
-    "id": "egg_macaw",
-    "name": "macaw egg",
-    "copy-from": "egg_bird",
-    "rot_spawn": "GROUP_EGG_MACAW",
-    "rot_spawn_chance": 70
+    "id": "egg_great_kiskadee",
+    "name": { "str": "great kiskadee egg" },
+    "copy-from": "egg_bluejay",
+    "rot_spawn": "GROUP_EGG_GREAT_KISKADEE"
+  },
+  {
+    "type": "COMESTIBLE",
+    "id": "egg_rufous_hornero",
+    "name": { "str": "rufous hornero egg" },
+    "copy-from": "egg_waxwing",
+    "rot_spawn": "GROUP_EGG_RUFOUS_HORNERO"
   },
   {
     "type": "COMESTIBLE",
@@ -104,14 +109,6 @@
     "rot_spawn": "GROUP_EGG_CURASSOW",
     "rot_spawn_chance": 70
   },
-  {
-    "type": "COMESTIBLE",
-    "id": "egg_flamingo",
-    "name": "flamingo egg",
-    "copy-from": "egg_bird",
-    "rot_spawn": "GROUP_EGG_FLAMINGO",
-    "rot_spawn_chance": 70
-  },
   {
     "type": "COMESTIBLE",
     "id": "egg_constrictor",
diff --git a/data/mods/TropiCataclysm/monstergroups/eggs.json b/data/mods/TropiCataclysm/monstergroups/eggs.json
index 215a19d308..06fa3fcb56 100644
--- a/data/mods/TropiCataclysm/monstergroups/eggs.json
+++ b/data/mods/TropiCataclysm/monstergroups/eggs.json
@@ -4,158 +4,135 @@
     "type": "monstergroup",
     "default": "mon_guinea_fowl_chick",
     "monsters": [
-      { "monster": "mon_guinea_fowl_chick", "weight": 50, "cost_multiplier": 1 },
-      { "monster": "mon_quail_chick", "weight": 100, "cost_multiplier": 1 },
-      { "monster": "mon_rhea_chick", "weight": 250, "cost_multiplier": 1 }
+      { "monster": "mon_guinea_fowl_chick", "weight": 1 },
+      { "monster": "mon_quail_chick", "weight": 2 },
+      { "monster": "mon_rhea_chick", "weight": 5 }
     ]
   },
   {
     "name": "GROUP_EGG_GUINEA_FOWL",
     "type": "monstergroup",
-    "default": "mon_guinea_fowl_chick",
-    "monsters": [ { "monster": "mon_guinea_fowl_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_guinea_fowl_chick" } ]
   },
   {
     "name": "GROUP_EGG_QUAIL",
     "type": "monstergroup",
-    "default": "mon_quail_chick",
-    "monsters": [ { "monster": "mon_quail_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_quail_chick" } ]
   },
   {
     "name": "GROUP_EGG_RHEA",
     "type": "monstergroup",
-    "default": "mon_rhea_chick",
-    "monsters": [ { "monster": "mon_rhea_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_rhea_chick" } ]
   },
   {
     "name": "GROUP_EGG_TOUCAN",
     "type": "monstergroup",
-    "default": "mon_toucan_chick",
-    "monsters": [ { "monster": "mon_toucan_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_toucan_chick" } ]
   },
   {
     "name": "GROUP_EGG_PARROT",
     "type": "monstergroup",
-    "default": "mon_parrot_chick",
-    "monsters": [ { "monster": "mon_parrot_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_parrot_chick" } ]
   },
   {
-    "name": "GROUP_EGG_BUDGERIGAR",
+    "name": "GROUP_EGG_MACAW",
     "type": "monstergroup",
-    "default": "mon_budgerigar_chick",
-    "monsters": [ { "monster": "mon_budgerigar_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_macaw_chick" } ]
   },
   {
-    "name": "GROUP_EGG_COCKATOO",
+    "name": "GROUP_EGG_FINCH_SAFFRON",
     "type": "monstergroup",
-    "default": "mon_cockatoo_chick",
-    "monsters": [ { "monster": "mon_cockatoo_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_finch_saffron_chick" } ]
   },
   {
-    "name": "GROUP_EGG_MACAW",
+    "name": "GROUP_EGG_GREAT_KISKADEE",
     "type": "monstergroup",
-    "default": "mon_macaw_chick",
-    "monsters": [ { "monster": "mon_macaw_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_great_kiskadee_chick" } ]
+  },
+  {
+    "name": "GROUP_EGG_RUFOUS_HORNERO",
+    "type": "monstergroup",
+    "monsters": [ { "monster": "mon_rufous_hornero_chick" } ]
   },
   {
     "name": "GROUP_EGG_TINAMOU",
     "type": "monstergroup",
-    "default": "mon_tinamou_chick",
-    "monsters": [ { "monster": "mon_tinamou_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_tinamou_chick" } ]
   },
   {
     "name": "GROUP_EGG_SERIEMA",
     "type": "monstergroup",
-    "default": "mon_seriema_chick",
-    "monsters": [ { "monster": "mon_seriema_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_seriema_chick" } ]
   },
   {
     "name": "GROUP_EGG_CHACHALACA",
     "type": "monstergroup",
-    "default": "mon_chachalaca_chick",
-    "monsters": [ { "monster": "mon_chachalaca_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_chachalaca_chick" } ]
   },
   {
     "name": "GROUP_EGG_GUAN",
     "type": "monstergroup",
-    "default": "mon_guan_chick",
-    "monsters": [ { "monster": "mon_guan_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_guan_chick" } ]
   },
   {
     "name": "GROUP_EGG_CURASSOW",
     "type": "monstergroup",
-    "default": "mon_curassow_chick",
-    "monsters": [ { "monster": "mon_curassow_chick", "weight": 100, "cost_multiplier": 1 } ]
-  },
-  {
-    "name": "GROUP_EGG_FLAMINGO",
-    "type": "monstergroup",
-    "default": "mon_flamingo_chick",
-    "monsters": [ { "monster": "mon_flamingo_chick", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_curassow_chick" } ]
   },
   {
     "name": "GROUP_EGG_CONSTRICTOR",
     "type": "monstergroup",
     "default": "mon_anaconda",
     "monsters": [
-      { "monster": "mon_anaconda", "weight": 40, "cost_multiplier": 1 },
-      { "monster": "mon_python", "weight": 20, "cost_multiplier": 1 },
-      { "monster": "mon_boa", "weight": 40, "cost_multiplier": 1 }
+      { "monster": "mon_anaconda", "weight": 2 },
+      { "monster": "mon_python", "weight": 1 },
+      { "monster": "mon_boa", "weight": 2 }
     ]
   },
   {
     "name": "GROUP_EGG_LIZARD",
     "type": "monstergroup",
-    "default": "mon_beaded_lizard",
-    "monsters": [ { "monster": "mon_beaded_lizard", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_beaded_lizard" } ]
   },
   {
     "name": "GROUP_EGG_MONITOR",
     "type": "monstergroup",
-    "default": "mon_monitor_lizard",
-    "monsters": [ { "monster": "mon_monitor_lizard", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_monitor_lizard" } ]
   },
   {
     "name": "GROUP_EGG_IGUANA",
     "type": "monstergroup",
-    "default": "mon_iguana",
-    "monsters": [ { "monster": "mon_iguana", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_iguana" } ]
   },
   {
     "name": "GROUP_EGG_TURTLE",
     "type": "monstergroup",
-    "default": "mon_turtle",
-    "monsters": [ { "monster": "mon_turtle", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_turtle" } ]
   },
   {
     "name": "GROUP_EGG_SNAPPER",
     "type": "monstergroup",
-    "default": "mon_turtle_snapper",
-    "monsters": [ { "monster": "mon_turtle_snapper", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_turtle_snapper" } ]
   },
   {
     "name": "GROUP_EGG_TORTOISE",
     "type": "monstergroup",
-    "default": "mon_tortoise",
-    "monsters": [ { "monster": "mon_tortoise", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_tortoise" } ]
   },
   {
     "name": "GROUP_EGG_TERMITE",
     "type": "monstergroup",
-    "default": "mon_termite_nymph",
-    "monsters": [ { "monster": "mon_termite_nymph", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_termite_nymph" } ]
   },
   {
     "name": "GROUP_EGG_TRIFFID",
     "type": "monstergroup",
-    "default": "mon_triffid_podling",
-    "monsters": [ { "monster": "mon_triffid_podling", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_triffid_podling" } ]
   },
   {
     "name": "GROUP_EGG_FUNGUS",
     "type": "monstergroup",
-    "default": "mon_fungaloid_saproling",
-    "monsters": [ { "monster": "mon_fungaloid_saproling", "weight": 100, "cost_multiplier": 1 } ]
+    "monsters": [ { "monster": "mon_fungaloid_saproling" } ]
   },
   {
     "name": "GROUP_EGG_SALAMANDER",
diff --git a/data/mods/TropiCataclysm/monstergroups/wilderness.json b/data/mods/TropiCataclysm/monstergroups/wilderness.json
index 4eafca1068..f5f2bb73a1 100644
--- a/data/mods/TropiCataclysm/monstergroups/wilderness.json
+++ b/data/mods/TropiCataclysm/monstergroups/wilderness.json
@@ -58,9 +58,10 @@
       { "monster": "mon_tortoise_small", "weight": 3, "pack_size": [ 1, 3 ], "starts": "240 hours" },
       { "monster": "mon_tortoise_giant", "weight": 3, "pack_size": [ 1, 3 ], "starts": "480 hours" },
       { "monster": "mon_parrot", "weight": 5, "cost_multiplier": 0, "pack_size": [ 1, 14 ] },
-      { "monster": "mon_budgerigar", "weight": 5, "cost_multiplier": 0, "pack_size": [ 1, 14 ] },
+      { "monster": "mon_finch_saffron", "weight": 5, "cost_multiplier": 0, "pack_size": [ 1, 3 ] },
+      { "monster": "mon_great_kiskadee", "weight": 5, "cost_multiplier": 0, "pack_size": [ 1, 2 ] },
+      { "monster": "mon_rufous_hornero", "weight": 5, "cost_multiplier": 0, "pack_size": [ 1, 3 ] },
       { "monster": "mon_toucan", "weight": 2, "cost_multiplier": 0, "pack_size": [ 1, 10 ] },
-      { "monster": "mon_cockatoo", "weight": 2, "cost_multiplier": 0, "pack_size": [ 1, 10 ] },
       { "monster": "mon_macaw", "weight": 3, "cost_multiplier": 0, "pack_size": [ 1, 8 ] },
       { "monster": "mon_seriema", "weight": 15, "cost_multiplier": 10 },
       { "monster": "mon_tinamou", "weight": 12, "cost_multiplier": 2, "pack_size": [ 1, 6 ] },
@@ -77,7 +78,6 @@
     "monsters": [
       { "monster": "mon_coypus", "weight": 15, "cost_multiplier": 10, "pack_size": [ 1, 3 ] },
       { "monster": "mon_turtle", "weight": 15, "cost_multiplier": 10, "pack_size": [ 1, 3 ] },
-      { "monster": "mon_flamingo", "weight": 15, "cost_multiplier": 5, "pack_size": [ 1, 4 ] },
       { "monster": "mon_capybara", "weight": 15, "cost_multiplier": 5, "pack_size": [ 1, 4 ] },
       { "monster": "mon_caiman", "weight": 10, "cost_multiplier": 5 },
       { "monster": "mon_crocodile", "weight": 10, "cost_multiplier": 5 },
@@ -142,27 +142,27 @@
   {
     "type": "monstergroup",
     "name": "GROUP_PARK_ANIMAL",
-    "//": "Current SPRING first DAY count is 425.  Note that 'freq' units are tenth of a percent, with default filling in the gap.",
     "is_animal": true,
     "monsters": [
-      { "monster": "mon_budgerigar", "weight": 35, "cost_multiplier": 0 },
       { "monster": "mon_parrot", "weight": 25, "cost_multiplier": 0 },
+      { "monster": "mon_finch_saffron", "weight": 25, "cost_multiplier": 0 },
+      { "monster": "mon_great_kiskadee", "weight": 25, "cost_multiplier": 0 },
+      { "monster": "mon_rufous_hornero", "weight": 25, "cost_multiplier": 0 },
       { "monster": "mon_iguana", "weight": 25, "cost_multiplier": 0 },
-      { "monster": "mon_cockatoo", "weight": 25, "cost_multiplier": 0 },
       { "monster": "mon_macaw", "weight": 15, "cost_multiplier": 0 }
     ]
   },
   {
     "type": "monstergroup",
     "name": "GROUP_ROOF_ANIMAL",
-    "//": "Current SPRING first DAY count is 350.  Note that 'freq' units are tenth of a percent, with default filling in the gap.",
     "is_animal": true,
     "monsters": [
-      { "monster": "mon_budgerigar", "weight": 45, "cost_multiplier": 0 },
       { "monster": "mon_parrot", "weight": 25, "cost_multiplier": 0 },
+      { "monster": "mon_finch_saffron", "weight": 25, "cost_multiplier": 0 },
+      { "monster": "mon_great_kiskadee", "weight": 20, "cost_multiplier": 0 },
+      { "monster": "mon_rufous_hornero", "weight": 20, "cost_multiplier": 0 },
       { "monster": "mon_boa", "weight": 25, "cost_multiplier": 0 },
       { "monster": "mon_gecko", "weight": 25, "cost_multiplier": 0 },
-      { "monster": "mon_cockatoo", "weight": 25, "cost_multiplier": 0 },
       { "monster": "mon_guinea_fowl", "weight": 10, "cost_multiplier": 0 },
       { "monster": "mon_macaw", "weight": 10, "cost_multiplier": 0 },
       {
@@ -191,22 +191,15 @@
       { "monster": "mon_anaconda", "weight": 40, "cost_multiplier": 0 }
     ]
   },
-  {
-    "type": "monstergroup",
-    "name": "GROUP_POND_BIRD",
-    "default": "mon_null",
-    "is_animal": true,
-    "monsters": [ { "monster": "mon_flamingo", "weight": 50, "cost_multiplier": 0 } ]
-  },
   {
     "type": "monstergroup",
     "name": "GROUP_BIRDFEEDER",
-    "//": "Current count is 225.  Note that 'freq' units are tenth of a percent, with default filling in the gap.",
     "is_animal": true,
     "monsters": [
       { "monster": "mon_parrot", "weight": 35, "cost_multiplier": 0 },
-      { "monster": "mon_cockatoo", "weight": 35, "cost_multiplier": 0 },
-      { "monster": "mon_budgerigar", "weight": 50, "cost_multiplier": 0 }
+      { "monster": "mon_finch_saffron", "weight": 20, "cost_multiplier": 0 },
+      { "monster": "mon_great_kiskadee", "weight": 15, "cost_multiplier": 0 },
+      { "monster": "mon_rufous_hornero", "weight": 15, "cost_multiplier": 0 }
     ]
   }
 ]
diff --git a/data/mods/TropiCataclysm/monsters/tropical_bird.json b/data/mods/TropiCataclysm/monsters/tropical_bird.json
index a8db6cf76d..81d9b21c5f 100644
--- a/data/mods/TropiCataclysm/monsters/tropical_bird.json
+++ b/data/mods/TropiCataclysm/monsters/tropical_bird.json
@@ -124,212 +124,121 @@
   {
     "id": "mon_toucan",
     "type": "MONSTER",
-    "name": { "str": "toucan" },
-    "description": "A small bird with dark plumage and a bright colored face and beak.  These tropical birds can be found all around the Americas feeding on fruits and seeds.",
-    "bodytype": "bird",
-    "default_faction": "small_animal",
-    "categories": [ "WILDLIFE" ],
-    "species": [ "BIRD" ],
-    "volume": "400 ml",
-    "weight": "400 g",
-    "hp": 10,
-    "speed": 140,
-    "material": [ "flesh" ],
-    "symbol": "v",
+    "name": { "str": "toco toucan" },
+    "description": "The largest species of toucan sporting dark plumage and a bright colored beak.  This bird can be found pretty much all over South America, feeding on fruits and seeds.",
+    "copy-from": "mon_bird_flying_base",
+    "volume": "867 ml",
+    "weight": "650 g",
+    "hp": 6,
     "color": "brown",
-    "aggression": -99,
-    "morale": -8,
-    "melee_dice": 1,
-    "melee_dice_sides": 1,
-    "melee_damage": [ { "damage_type": "cut", "amount": 0 } ],
-    "dodge": 4,
-    "harvest": "bird_small",
-    "fear_triggers": [ "SOUND", "PLAYER_CLOSE", "FRIEND_ATTACKED", "FRIEND_DIED", "FIRE", "HURT" ],
-    "reproduction": { "baby_egg": "egg_toucan", "baby_count": 5, "baby_timer": 14 },
-    "baby_flags": [ "SPRING" ],
-    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 8 },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER", "WARM", "FLIES", "SWARMS" ]
+    "reproduction": { "baby_egg": "egg_toucan", "baby_count": 4, "baby_timer": 63 }
   },
   {
-    "id": "mon_parrot",
+    "id": "mon_toucan_chick",
     "type": "MONSTER",
-    "name": { "str": "parrot" },
-    "description": "A small to medium bird with bright colored plumage and a short hooked beak.  These birds are considered to be quite intelligent and have been domesticated heavily as pets.",
-    "bodytype": "bird",
-    "default_faction": "small_animal",
-    "categories": [ "WILDLIFE" ],
-    "species": [ "BIRD" ],
-    "volume": "400 ml",
-    "weight": "400 g",
-    "hp": 10,
-    "speed": 140,
-    "material": [ "flesh" ],
-    "symbol": "v",
-    "color": "brown",
-    "aggression": -99,
-    "morale": -8,
-    "melee_dice": 1,
-    "melee_dice_sides": 1,
-    "melee_damage": [ { "damage_type": "cut", "amount": 0 } ],
-    "dodge": 4,
-    "harvest": "bird_small",
-    "fear_triggers": [ "SOUND", "PLAYER_CLOSE", "FRIEND_ATTACKED", "FRIEND_DIED", "FIRE", "HURT" ],
-    "reproduction": { "baby_egg": "egg_parrot", "baby_count": 5, "baby_timer": 14 },
-    "baby_flags": [ "SPRING" ],
-    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 8 },
-    "petfood": {
-      "food": [ "BIRDFOOD" ],
-      "feed": "The %s seems to like you!  It runs around your legs and seems friendly.",
-      "pet": "The %s runs around your leg."
-    },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER", "WARM", "FLIES", "SWARMS", "CANPLAY" ]
+    "copy-from": "mon_generic_chick",
+    "upgrades": { "age_grow": 35, "into": "mon_toucan" },
+    "//": "Copied from the vanilla corvids. ~8 weeks to fledge, but couldn't find any info on how long it takes for them to reach adult size. Sexual maturity is a bad metric as they reach it very late, at 3-4 years of age."
   },
   {
-    "id": "mon_cockatoo",
+    "id": "mon_parrot",
     "type": "MONSTER",
-    "name": { "str": "cockatoo" },
-    "description": "A small to medium bird with bright colored plumage and a short hooked beak.  These relatives of parrots originate from Indonesia and Oceania, and are well known for their antics and head plumage.",
-    "bodytype": "bird",
-    "default_faction": "small_animal",
-    "categories": [ "WILDLIFE" ],
-    "species": [ "BIRD" ],
-    "volume": "400 ml",
-    "weight": "400 g",
-    "hp": 10,
-    "speed": 140,
-    "material": [ "flesh" ],
-    "symbol": "v",
-    "color": "brown",
-    "aggression": -99,
-    "morale": -8,
-    "melee_dice": 1,
-    "melee_dice_sides": 1,
-    "melee_damage": [ { "damage_type": "cut", "amount": 0 } ],
-    "dodge": 4,
-    "harvest": "bird_small",
-    "fear_triggers": [ "SOUND", "PLAYER_CLOSE", "FRIEND_ATTACKED", "FRIEND_DIED", "FIRE", "HURT" ],
-    "reproduction": { "baby_egg": "egg_cockatoo", "baby_count": 5, "baby_timer": 14 },
-    "baby_flags": [ "SPRING" ],
-    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 8 },
-    "petfood": {
-      "food": [ "BIRDFOOD" ],
-      "feed": "The %s seems to like you!  It runs around your legs and seems friendly.",
-      "pet": "The %s runs around your leg."
-    },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER", "WARM", "FLIES", "SWARMS", "CANPLAY" ]
+    "name": { "str": "white-eyed parakeet" },
+    "description": "A small bird with bright green-colored plumage and a short hooked beak.  This species is widespread throughout all of South America, on top of being quite intelligent and sometimes kept as pets.",
+    "copy-from": "mon_bird_flying_base",
+    "volume": "220 ml",
+    "weight": "165 g",
+    "color": "light_green",
+    "reproduction": { "baby_egg": "egg_parrot", "baby_count": 3, "baby_timer": 38 }
   },
   {
-    "id": "mon_budgerigar",
+    "id": "mon_parrot_chick",
     "type": "MONSTER",
-    "name": { "str": "budgerigar" },
-    "description": "A small bird with with bright colored plumage and a short hooked beak.  These small relatives of the parrot come from Australia but were domesticated worldwide as pets.",
-    "bodytype": "bird",
-    "default_faction": "small_animal",
-    "categories": [ "WILDLIFE" ],
-    "species": [ "BIRD" ],
-    "volume": "180 ml",
-    "weight": "35 g",
-    "hp": 5,
-    "speed": 140,
-    "material": [ "flesh" ],
-    "symbol": "v",
-    "color": "brown",
-    "aggression": -99,
-    "morale": -8,
-    "melee_dice": 1,
-    "melee_dice_sides": 1,
-    "melee_damage": [ { "damage_type": "cut", "amount": 0 } ],
-    "dodge": 4,
-    "harvest": "bird_tiny",
-    "fear_triggers": [ "SOUND", "PLAYER_CLOSE", "FRIEND_ATTACKED", "FRIEND_DIED", "FIRE", "HURT" ],
-    "reproduction": { "baby_egg": "egg_budgerigar", "baby_count": 5, "baby_timer": 14 },
-    "baby_flags": [ "SPRING" ],
-    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 8 },
+    "copy-from": "mon_generic_chick_tiny",
+    "upgrades": { "age_grow": 12, "into": "mon_parrot" },
+    "//": "12 days to reach independence. Use the default multiplier of 4 to our advantage to get a rough estimate of reaching adult size.",
     "petfood": {
       "food": [ "BIRDFOOD" ],
       "feed": "The %s seems to like you!  It runs around your legs and seems friendly.",
       "pet": "The %s runs around your leg."
-    },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER", "WARM", "FLIES", "SWARMS", "CANPLAY" ]
+    }
   },
   {
     "id": "mon_macaw",
     "type": "MONSTER",
-    "name": { "str": "macaw" },
-    "description": "A medium to large bird with bright colored plumage and a short hooked beak.  These are the largest flying parrots in the world, living usually in the thickest rainforests.",
-    "bodytype": "bird",
-    "default_faction": "small_animal",
-    "categories": [ "WILDLIFE" ],
-    "species": [ "BIRD" ],
-    "volume": "1000 ml",
-    "weight": "900 g",
-    "hp": 25,
-    "speed": 140,
-    "material": [ "flesh" ],
-    "symbol": "v",
-    "color": "brown",
-    "aggression": -99,
-    "morale": -8,
-    "melee_dice": 1,
-    "melee_dice_sides": 1,
-    "melee_damage": [ { "damage_type": "cut", "amount": 0 } ],
-    "dodge": 4,
-    "harvest": "bird_small",
-    "fear_triggers": [ "SOUND", "PLAYER_CLOSE", "FRIEND_ATTACKED", "FRIEND_DIED", "FIRE", "HURT" ],
-    "reproduction": { "baby_egg": "egg_macaw", "baby_count": 5, "baby_timer": 14 },
-    "baby_flags": [ "SPRING" ],
-    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 8 },
+    "name": { "str": "blue-and-yellow macaw" },
+    "description": "A large parrot sporting beautiful blue plumage.  Intelligent and often kept as a pet.",
+    "copy-from": "mon_bird_flying_base",
+    "volume": "1800 ml",
+    "weight": "1350 g",
+    "hp": 8,
+    "color": "light_blue",
+    "reproduction": { "baby_egg": "egg_macaw", "baby_count": 3, "baby_timer": 116 }
+  },
+  {
+    "id": "mon_macaw_chick",
+    "type": "MONSTER",
+    "copy-from": "mon_generic_chick",
+    "upgrades": { "age_grow": 23, "into": "mon_macaw" },
+    "//": "3 months to reach independence.",
     "petfood": {
       "food": [ "BIRDFOOD" ],
       "feed": "The %s seems to like you!  It runs around your legs and seems friendly.",
       "pet": "The %s runs around your leg."
-    },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER", "WARM", "FLIES", "SWARMS", "CANPLAY" ]
+    }
   },
   {
-    "id": "mon_toucan_chick",
+    "id": "mon_finch_saffron",
     "type": "MONSTER",
-    "copy-from": "mon_crow_chick",
-    "name": { "str": "brown chick" },
-    "volume": "200 ml",
-    "weight": "15 g",
-    "upgrades": { "age_grow": 11, "into": "mon_toucan" }
+    "name": { "str": "saffron finch", "str_pl": "saffron finches" },
+    "description": "A tiny, golden bird bopping around in the canopy of trees.  Its cone-shaped bill tells you it s a seed eater.",
+    "copy-from": "mon_bird_flying_base",
+    "color": "yellow",
+    "volume": "27 ml",
+    "weight": "20 g",
+    "reproduction": { "baby_egg": "egg_finch_saffron", "baby_count": 5, "baby_timer": 16 },
+    "//": "Values for reproduction and chick growth copied from the cedar waxwing, as no info could be found online."
   },
   {
-    "id": "mon_parrot_chick",
+    "id": "mon_finch_saffron_chick",
     "type": "MONSTER",
-    "copy-from": "mon_crow_chick",
-    "name": { "str": "brown chick" },
-    "volume": "200 ml",
-    "weight": "15 g",
-    "upgrades": { "age_grow": 11, "into": "mon_parrot" }
+    "copy-from": "mon_generic_chick_tiny",
+    "upgrades": { "age_grow": 8, "into": "mon_finch_saffron" }
   },
   {
-    "id": "mon_budgerigar_chick",
+    "id": "mon_great_kiskadee",
     "type": "MONSTER",
-    "copy-from": "mon_crow_chick",
-    "name": { "str": "brown chick" },
-    "volume": "160 ml",
-    "weight": "10 g",
-    "upgrades": { "age_grow": 11, "into": "mon_budgerigar" }
+    "name": { "str": "great kiskadee" },
+    "description": "This large flycatcher has a black crown, prominent white eye stripe and a dominantly yellow body.  At almost a foot long, it swoops up a variety of flying insects.",
+    "copy-from": "mon_bird_flying_base",
+    "color": "yellow",
+    "volume": "80 ml",
+    "weight": "60 g",
+    "reproduction": { "baby_egg": "egg_great_kiskadee", "baby_count": 5, "baby_timer": 18 },
+    "//": "Values for reproduction and chick growth copied from the bluejay, as no info could be found online."
   },
   {
-    "id": "mon_cockatoo_chick",
+    "id": "mon_great_kiskadee_chick",
     "type": "MONSTER",
-    "copy-from": "mon_crow_chick",
-    "name": { "str": "brown chick" },
-    "volume": "200 ml",
-    "weight": "15 g",
-    "upgrades": { "age_grow": 11, "into": "mon_cockatoo" }
+    "copy-from": "mon_generic_chick_tiny",
+    "upgrades": { "age_grow": 10, "into": "mon_great_kiskadee" }
   },
   {
-    "id": "mon_macaw_chick",
+    "id": "mon_rufous_hornero",
     "type": "MONSTER",
-    "copy-from": "mon_crow_chick",
-    "name": { "str": "brown chick" },
-    "volume": "240 ml",
-    "weight": "20 g",
-    "upgrades": { "age_grow": 11, "into": "mon_macaw" }
+    "name": { "str": "rufous hornero" },
+    "description": "A drab, robin-sized ground bird with startling red eyes.  More often spotted are their odd oven shaped nests.",
+    "copy-from": "mon_bird_flying_base",
+    "volume": "60 ml",
+    "weight": "45 g",
+    "reproduction": { "baby_egg": "egg_rufous_hornero", "baby_count": 4, "baby_timer": 44 }
+  },
+  {
+    "id": "mon_rufous_hornero_chick",
+    "type": "MONSTER",
+    "copy-from": "mon_generic_chick_tiny",
+    "upgrades": { "age_grow": 25, "into": "mon_rufous_hornero" },
+    "//": "24-26 days to reach independence. Use the default multiplier of 4 to our advantage to get a rough estimate of reaching adult size."
   },
   {
     "id": "mon_tinamou",
@@ -460,45 +369,5 @@
     "copy-from": "mon_chicken_chick",
     "name": { "str": "brown chick" },
     "upgrades": { "age_grow": 10, "into": "mon_seriema" }
-  },
-  {
-    "id": "mon_flamingo",
-    "type": "MONSTER",
-    "name": { "str": "flamingo" },
-    "description": "A medium to large bird with white and pink plumage.  They can be found on both hemispheres living around shallow waters rich on plankton and small crustaceans.",
-    "default_faction": "small_animal",
-    "bodytype": "bird",
-    "categories": [ "WILDLIFE" ],
-    "species": [ "BIRD" ],
-    "volume": "28 L",
-    "weight": "2800 g",
-    "hp": 30,
-    "speed": 140,
-    "material": [ "flesh" ],
-    "symbol": "f",
-    "color": "pink",
-    "aggression": -10,
-    "morale": 25,
-    "melee_dice": 1,
-    "melee_dice_sides": 1,
-    "melee_damage": [ { "damage_type": "cut", "amount": 1 } ],
-    "dodge": 3,
-    "harvest": "bird_large",
-    "vision_day": 50,
-    "fear_triggers": [ "PLAYER_CLOSE", "FRIEND_DIED", "FIRE", "HURT" ],
-    "reproduction": { "baby_egg": "egg_flamingo", "baby_count": 1, "baby_timer": 10 },
-    "biosignature": { "biosig_item": "feces_bird", "biosig_timer": 8 },
-    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER", "WARM", "FLIES", "SWARMS" ]
-  },
-  {
-    "id": "mon_flamingo_chick",
-    "type": "MONSTER",
-    "copy-from": "mon_duck_chick",
-    "name": { "str": "flamingo chick" },
-    "description": "A baby flamingo covered in fuzzy plumage.  Unlike their parents they are born gray or white and obtain their pink color as they grow feeding on crustaceans.",
-    "volume": "280 ml",
-    "weight": "280 g",
-    "color": "light_gray",
-    "upgrades": { "age_grow": 16, "into": "mon_flamingo" }
   }
 ]
diff --git a/data/mods/TropiCataclysm/obsoletion_and_migration_0.I/migration_items.json b/data/mods/TropiCataclysm/obsoletion_and_migration_0.I/migration_items.json
new file mode 100644
index 0000000000..aea58015da
--- /dev/null
+++ b/data/mods/TropiCataclysm/obsoletion_and_migration_0.I/migration_items.json
@@ -0,0 +1,17 @@
+[
+  {
+    "id": "egg_budgerigar",
+    "type": "MIGRATION",
+    "replace": "egg_parrot"
+  },
+  {
+    "id": "egg_cockatoo",
+    "type": "MIGRATION",
+    "replace": "egg_parrot"
+  },
+  {
+    "id": "egg_flamingo",
+    "type": "MIGRATION",
+    "replace": "egg_duck"
+  }
+]
diff --git a/data/mods/TropiCataclysm/requirements.json b/data/mods/TropiCataclysm/requirements.json
new file mode 100644
index 0000000000..c36166ca23
--- /dev/null
+++ b/data/mods/TropiCataclysm/requirements.json
@@ -0,0 +1,18 @@
+[
+  {
+    "id": "eggs_bird",
+    "type": "requirement",
+    "extend": {
+      "components": [
+        [
+          [ "egg_toucan", 3 ],
+          [ "egg_parrot", 6 ],
+          [ "egg_macaw", 3 ],
+          [ "egg_rufous_hornero", 20 ],
+          [ "egg_finch_saffron", 20 ],
+          [ "egg_great_kiskadee", 6 ]
+        ]
+      ]
+    }
+  }
+]
diff --git a/tools/spell_checker/dictionary.txt b/tools/spell_checker/dictionary.txt
index 6e49c8a34b..0ee7880372 100644
--- a/tools/spell_checker/dictionary.txt
+++ b/tools/spell_checker/dictionary.txt
@@ -2676,6 +2676,8 @@ Horayot
 horchata
 horks
 Hornblower
+hornero
+horneros
 Horrorpelta
 Horrorpeltas
 hosomaki
@@ -3158,6 +3160,8 @@ kippahs
 Kippur
 kirpan
 kirpans
+kiskadee
+kiskadees
 Kitáb
 KITT
 kittel
@@ -6239,6 +6243,7 @@ tlatoani
 TNT
 Toby
 tock
+toco
 TODO
 Toffer
 toggleable
-- 
2.45.2

